src/phoenix/schemas/core.py

```diff
@@ -0,0 +1,62 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Dict, Optional
+
+from phoenix.db import models
+from sqlalchemy import case, func, select
+from sqlalchemy.ext.asyncio import AsyncSession
+
+
+@dataclass
+class Dataset:
+    id: int
+    name: str
+    description: Optional[str]
+    created_at: datetime
+    updated_at: datetime
+    metadata: Dict[str, Any]
+
+    @classmethod
+    def from_model(cls, model: models.Dataset) -> "Dataset":
+        return cls(
+            id=model.id,
+            name=model.name,
+            description=model.description,
+            created_at=model.created_at,
+            updated_at=model.updated_at,
+            metadata=model.metadata_,
+        )
+
+    async def count_active_examples(self, session: AsyncSession) -> int:
+        result = await session.execute(
+            select(
+                func.sum(
+                    case(
+                        (models.DatasetExampleRevision.revision_kind == "CREATE", 1),
+                        (models.DatasetExampleRevision.revision_kind == "DELETE", -1),
+                        else_=0,
+                    )
+                )
+            )
+            .select_from(models.DatasetExampleRevision)
+            .join(
+                models.DatasetExample,
+                models.DatasetExample.id == models.DatasetExampleRevision.dataset_example_id,
+            )
+            .filter(models.DatasetExample.dataset_id == self.id)
+        )
+        active_count = result.scalar()
+        assert isinstance(active_count, int) or active_count is None
+        return active_count if active_count is not None else 0
+
+    async def serialize(self, session: AsyncSession) -> Dict[str, Any]:
```

Comment:
Feels a little unexpected to have a serialize method that needs to make a db query to serialize an object.

Reply:
we will refactor this but ultimately more SQL needs to be run than a simple query for an ID because we do not store example counts in the DB